### PR TITLE
AB test GPT lazy loading

### PR DESCRIPTION
**Pull request by zekehuntergreen**
## What does this change?
- Introduces an AB test to compare our current implementation of lazily-loaded ads to [GPT's](https://developers.google.com/publisher-tag/reference#googletag.PubAdsService_enableLazyLoad) 
- A [very similar test](https://github.com/guardian/frontend/pull/22327) was done a few years ago and concluded that GPT's lazy loading did not improve the metrics that they were looking at. That test used the default values for the margin at which ads were fetched and rendered.
- The options passed to `enableLazyLoad` let us specify a margin at which ads are loaded as a percentage of the viewport height, whereas our current system loads ads when they are 200px away from the viewport. In order to try to make as fair a comparison as possible, we've approximated that 200px as a percentage of the average screen height (1080px) is 18.5%.
- Whether or not the test reveals that GPT's lazy loading system is an improvement, we're planning to do another test to compare different values for the margin.

## Does this change need to be reproduced in dotcom-rendering ?

- [ ] No
- [x] Yes [(please indicate your plans for DCR Implementation)](https://github.com/guardian/dotcom-rendering/pull/4387)

## What is the value of this and can you measure success?
- We suspect GPT's implementation of lazy loading will lead to higher ad ratio and / or viewability, and lower CLS.
- If these two metrics stay the same, using GPT's lazy loading could still be used instead of our custom implementation in order to simplify our code base.

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [x] On CODE (optional)

### CODE REVIEW

File: static/src/javascripts/projects/commercial/modules/dfp/lazy-load.ts [hunk 1]
```diff
@@ -43,7 +45,11 @@ const getObserver = once(() =>
 );
 
 export const enableLazyLoad = (advert: Advert): void => {
-	if (dfpEnv.lazyLoadObserve) {
+	const useGptLazyLoad = isInVariantSynchronous(
+		commercialGptLazyLoad,
+		'variant',
+	);
+	if (dfpEnv.lazyLoadObserve && !useGptLazyLoad) {
```
Comment (arelra) [inline static/src/javascripts/projects/commercial/modules/dfp/lazy-load.ts:52]: For readability could we extract this condition to something like `useCustomIntersectionObserver`?

```suggestion
	const useCustomLazyLoad = dfpEnv.lazyLoadObserve && !useGptLazyLoad;
	if (useCustomLazyLoad) {
```

File: static/src/javascripts/projects/commercial/modules/dfp/display-lazy-ads.ts [hunk 2]
```diff
@@ -1,11 +1,33 @@
...
 const instantLoadAdvertIds = ['dfp-ad--im'];
 
 const displayLazyAds = (): void => {
+	const useGptLazyLoad = isInVariantSynchronous(
+		commercialGptLazyLoad,
+		'variant',
+	);
+	if (useGptLazyLoad) {
+		window.googletag.pubads().enableLazyLoad({
+			// average mobile and desktop screen size ~ 1080px
```
Comment (arelra) [inline static/src/javascripts/projects/commercial/modules/dfp/display-lazy-ads.ts:20]: Pedantry :)

```suggestion
			// average mobile and desktop viewport height ~ 1080px
```

File: static/src/javascripts/projects/commercial/modules/dfp/load-advert.ts [hunk 3]
```diff
@@ -32,6 +32,7 @@ const eventTimer = EventTimer.get();
 
 export const loadAdvert = (advert: Advert): void => {
 	const adName = stripDfpAdPrefixFrom(advert.id);
+	// TODO can slotReady come after header bidding?
```
Comment (arelra) [inline static/src/javascripts/projects/commercial/modules/dfp/load-advert.ts:35]: If we are leaving this comment in could we add a bit more context pls around the implication of the answer?

File: static/src/javascripts/projects/commercial/modules/dfp/display-lazy-ads.ts [hunk 4]
```diff
@@ -1,11 +1,33 @@
+import { isInVariantSynchronous } from 'common/modules/experiments/ab';
+import { commercialGptLazyLoad } from 'common/modules/experiments/tests/commercial-gpt-lazy-load';
 import { partition } from '../../../../lib/partition';
 import { dfpEnv } from './dfp-env';
 import { enableLazyLoad } from './lazy-load';
 import { loadAdvert } from './load-advert';
 
+// For users in variant group of commercialGptLazyLoad,
+// these ad slots will be be lazy loaded rather than loaded instantly.
+// TODO - remove if GPT lazy loading is fully rolled out.
```
Comment (arelra) [inline static/src/javascripts/projects/commercial/modules/dfp/display-lazy-ads.ts:10]: Can we track differences in CLS _just_ for pages with "im" slots with the new lazy loading?

If we average out across all pages, I worry we might lose some CLS signal just for "im" ?

File: static/src/javascripts/projects/commercial/modules/dfp/display-lazy-ads.ts (hunk 2 above)
Comment (zekehuntergreen) [reply static/src/javascripts/projects/commercial/modules/dfp/display-lazy-ads.ts:20]: We actually only have approximate screen size which we're using as a proxy for viewport height! 
I'll update the comment to explain that better.

File: static/src/javascripts/projects/commercial/modules/dfp/load-advert.ts (hunk 3 above)
Comment (arelra) [reply static/src/javascripts/projects/commercial/modules/dfp/load-advert.ts:35]: Great thanks!

File: static/src/javascripts/projects/commercial/modules/dfp/display-lazy-ads.ts (hunk 4 above)
Comment (zekehuntergreen) [reply static/src/javascripts/projects/commercial/modules/dfp/display-lazy-ads.ts:10]: Good call. I've asked Yen and Sara about this.

**Review by simonbyford — APPROVED** [review]
Looks good!

**Review by arelra — APPROVED** [review]
👍

File: static/src/javascripts/projects/commercial/modules/dfp/display-lazy-ads.ts (hunk 4 above)
Comment (zekehuntergreen) [reply static/src/javascripts/projects/commercial/modules/dfp/display-lazy-ads.ts:10]: With Sara we decided we'd collect core web vitals data for all users participating in the test in order to make sure that pages with an `im` slot don't see an increase in CLS.
We'll need to merge https://github.com/guardian/dotcom-rendering/pull/4288 before setting this test live.

**Pull request merged**